Add type annotation to numpy method in BaseResourceVariable (Fixes #99580)

zvoicu000 opened the pull request:
This PR adds a type annotation to the numpy method in TensorFlow’s BaseResourceVariable class, specifying it returns a np.ndarray.
This resolves the static type checking issue reported in #99580.
Fixes #99580

zvoicu000 committed:
Add type annotation to numpy method in BaseResourceVariable (Fixes #99580)

zvoicu000 committed:
Fix line too long for Pylint compliance

mihaimaruseac reviewed (changes requested):
It doesn't make sense to add type annotations for just one single method.

zvoicu000 committed:
Add type annotations to all methods and properties
in BaseResourceVariable for consistency; improves static type checking,
addresses reviewer feedback, and fixes #99580.
